### ISSUE TITLE
FileExplorer: hidden drives are displayed 

### ISSUE THREAD

**Issue by jp-weber** [SafeExamBrowser/seb-win-refactoring#28]
**Describe the Bug**
The new explorer adds all drives which are connected but even hidden drives are displayed in the explorer.

**Steps to Reproduce**
1. Select e.g. a network drive and set 'NoDrives' for hiding, e.g. https://winaero.com/blog/hide-drive-windows-10-file-explorer/

2. Open SEB and select the file explorer.
4. See all drives.

**Expected Behavior**
Only visible drives in the Windows Explorer should be accessable.

**Version Information**
 - OS: Windows 10 Enterprise LTSC 1809
 - SEB-Version SEB 3.0

**Comment by dbuechel** [comment]
Can you specify how and where to set "NoDrive"? I fail to reproduce your issue.

**Comment by jp-weber** [comment]
To hide a drive it is necessary to set a registry entry "NoDrives" (also via GPO):
https://winaero.com/blog/hide-drive-windows-10-file-explorer/

So that the file explorer of the SEB can also implement this, it would be necessary to query whether NoDrives exists under the Explorer in the registry and which value is set.

**Comment by jp-weber** [comment]
@dbuechel  We have fixed this issue in our fork in FileSystemDialog.xaml.cs and commented our edits with 'uni-goettingen-patch: start' and 'uni-goettingen-patch: end'

Should we create a pull, or do you want to add it directly?

```csharp

// uni-goettingen-patch: start
using Microsoft.Win32;
// uni-goettingen-patch: end

namespace SafeExamBrowser.UserInterface.Desktop.Windows
{
	internal partial class FileSystemDialog : Window
	{
		// uni-goettingen-patch: begin
		private DriveInfo[] GetDrives(bool showAll = false)
		{
			var drives = DriveInfo.GetDrives();
			int noDrives = (int)Registry.GetValue("HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\Policies\\Explorer", "NoDrives", 0);

			if (noDrives > 0 || showAll)
			{
				return drives.Where(drive => (noDrives & (int)(Math.Pow(2, (int)(drive.RootDirectory.ToString()[0]) - 65))) == 0).ToArray();
			}

			return drives;
		}
		// uni-goettingen-patch: end

		private void InitializeFileSystem()
		{
			// uni-goettingen-patch: begin
			foreach (var drive in GetDrives())
			// uni-goettingen-patch: end
			{
				FileSystem.Items.Add(CreateItem(drive.RootDirectory));
			}

			if (FileSystem.HasItems && FileSystem.Items[0] is TreeViewItem item)
			{
				item.IsSelected = true;
			}

			if (!string.IsNullOrEmpty(initialPath))
			{
				var root = Path.GetPathRoot(initialPath);
				var path = initialPath.Replace(root, "").Split(Path.DirectorySeparatorChar);
				var segments = new List<string> { root };

				segments.AddRange(path);

				SelectInitialPath(FileSystem.Items, segments);

				if (element == FileSystemElement.File && operation == FileSystemOperation.Save)
				{
					NewElementName.Text = Path.GetFileName(initialPath);
				}
			}
		}
	}
}
```

**Comment by dbuechel** [comment]
Ah, great! Yes, please create a pull request.

**Comment by jp-weber** [comment]
Fixed with pull request #29
